packages/flame/lib/src/game/mixins/game.dart

```diff
@@ -42,8 +42,32 @@ mixin Game {
   Vector2? _size;
 
   /// This variable ensures that Game's [onLoad] is called no more than once.
+  Future<void>? _onLoadFuture;
+
+  bool _debugOnLoadStarted = false;
+
   @internal
-  late Future<void>? onLoadFuture = onLoad();
+  Future<void>? startLoad() {
+    assert(
+      () {
+        _debugOnLoadStarted = true;
+        return true;
+      }(),
+    );
+    return _onLoadFuture = onLoad();
```

Comment:
This needs to be stored outside of the method right? Because we should only load before the first time a component is mounted.

Reply:
And if it is stored outside of the method, this method doesn't really do anything anymore right? Except setting `_debugOnLoadStarted` to true, but it doesn't actually know that the called of this method awaits the future.

Reply:
I guess since it `onLoadFuture` could be internal we could guarantee that ourselves though.